api/odl.py

```diff
@@ -1085,15 +1085,32 @@ def checkout(self, patron, pin, licensepool, internal_format):
             raise CannotLoan()
         external_identifier = info_links[0].get("href")
 
-        return LoanInfo(
-            licensepool.collection,
-            licensepool.data_source.name,
-            licensepool.identifier.type,
-            licensepool.identifier.identifier,
-            start,
-            end,
-            external_identifier=external_identifier
-        )
+        if availability.get("status") == "available":
+            return LoanInfo(
+                licensepool.collection,
+                licensepool.data_source.name,
+                licensepool.identifier.type,
+                licensepool.identifier.identifier,
+                start,
+                end,
+                external_identifier=external_identifier
+            )
+        else:
+            # We tried to borrow this book but it wasn't available,
+            # so we got a hold.
+            position = entry.get("opds_holds", {}).get("position")
```

Comment:
Is this the only other option? Could there be a server that just doesn't support holds? Or in that case would the remote raise an exception before execution go to this point?

Reply:
I haven't provided a way to disable holds for these, but it might be a good idea to explicitly check for 'reserved' status just in case.